### ISSUE TITLE
Offline: Vagrant plugin install 

### ISSUE THREAD

**Issue by berndschatz** [hashicorp/vagrant#12593]
Is it possible to install a plugin offline ?

```bash
$ vagrant plugin install ./vagrant-disksize-0.1.3.gem  --local
Installing the './vagrant-disksize-0.1.3.gem' plugin. This can take a few minutes...
Vagrant failed to load a configured plugin source. This can be caused
by a variety of issues including: transient connectivity issues, proxy
filtering rejecting access to a configured plugin source, or a configured
plugin source not responding correctly. Please review the error message
below to help resolve the issue:

  timed out (https://gems.hashicorp.com/specs.4.8.gz)

Source: https://gems.hashicorp.com/
```

The gem has no dependencies so there is no need to access gems.hashicorp.com !?

**Comment by Ebsan** [comment]
I'm having the same issue and can't install a plugin when offline. For me If I put use the `--debug` argument as well the ultimate error is:

`no such name (https://gems.hashicorp.com/specs.4.8.gz)`

**Comment by berndschatz** [comment]
Still not working:

$ vagrant plugin install vagrant-disksize-0.1.3.gem  --local
Installing the 'vagrant-disksize-0.1.3.gem' plugin. This can take a few minutes...
Vagrant failed to load a configured plugin source. This can be caused
by a variety of issues including: transient connectivity issues, proxy
filtering rejecting access to a configured plugin source, or a configured
plugin source not responding correctly. Please review the error message
below to help resolve the issue:

  no such name (https://gems.hashicorp.com/specs.4.8.gz)

Source: https://gems.hashicorp.com/


beschat@bullseye:~/Vagrant/bionic64$ vagrant --version
Vagrant 2.2.14

**Comment by edpimenta** [comment]
Same here, with some debug info:
---
...
DEBUG bundler: Enabling user defined remote RubyGems sources
DEBUG bundler: Enabling default remote RubyGems sources
DEBUG bundler: Adding source - https://gems.hashicorp.com/
DEBUG bundler: Adding source - https://rubygems.org/
ERROR bundler: Failed to load configured plugin source `#<Gem::Source:0x00000000034c1370>`: no such name (https://gems.hashicorp.com/specs.4.8.gz)
ERROR warden: Error occurred: Vagrant failed to load a configured plugin source. This can be caused
by a variety of issues including: transient connectivity issues, proxy
filtering rejecting access to a configured plugin source, or a configured
plugin source not responding correctly. Please review the error message
below to help resolve the issue:

  no such name (https://gems.hashicorp.com/specs.4.8.gz)

**Comment by berndschatz** [comment]
> ## In the end it was due to network configuration at the company VPN. I have bypassed it (at home) and it worked.



[...]
> ## In the end it was due to network configuration at the company VPN. I have bypassed it (at home) and it worked.

@edpimenta this issue is about to install a plugin **offline**. 
Therefore: are you able to install a plugin without network connection ?

**Comment by edpimenta** [comment]
No, @berndschatz. I was trying to do that inside a closed VPN with restricted external access, but it only worked when I disabled it to be able to reach rubygems.org. 
So the short answer to your question so far seems to be "no - plugin installation seems to need the network always". At least I was not able to succeed otherwise.

**Comment by Abyss-W4tcher** [comment]
I've found a way to do this, for the vagrant-vbguest plugin:

On a machine with Internet:
```sh
gem fetch vagrant-vbguest -v 0.32.0
gem fetch micromachine -v 3.0.0
```

Transfer the files to the local machine, then:
```sh
# If a dependency is missing, this will fail and tell you which one to install
VAGRANT_ALLOW_PLUGIN_SOURCE_ERRORS=1 vagrant plugin install /tmp/micromachine-3.0.0.gem
VAGRANT_ALLOW_PLUGIN_SOURCE_ERRORS=1 vagrant plugin install /tmp/vagrant-vbguest-0.32.0.gem
```
